shared/chat/inbox/row/bottom-line.js

```diff
@@ -0,0 +1,140 @@
+// @flow
+import React from 'react'
+import {Text, Markdown, Box} from '../../../common-adapters'
+import {
+  globalStyles,
+  globalColors,
+  globalMargins,
+  styleSheetCreate,
+  collapseStyles,
+  lineHeight,
+} from '../../../styles'
+import {isMobile} from '../../../constants/platform'
+
+type BottomLineProps = {
+  backgroundColor: ?string,
+  participantNeedToRekey: boolean,
+  showBold: boolean,
+  snippet: ?string,
+  subColor: ?string,
+  youNeedToRekey: boolean,
+}
+
+function BottomLine(props: BottomLineProps) {
```

Comment:
Purecomponent has a shallow equal compare of props and state so if those === it'll not rerender. Currently functions don't do this check (even though it's safe for us). We could wrap with recompose and it has a pure helper or we can just extend from pure component. 

https://github.com/facebook/react/issues/5677